### DISCUSSION TITLE
EFX Toolkit to validate XML files with Business Rules

### DISCUSSION THREAD

**Discussion by SkySoul95** [OP-TED/eForms-SDK#348] [Q&A] [answered]
Hello and sorry for my bad english.
I'd like to ask you what could be the correct way to implement a solution in which XML files are already filled and the software checks if each BT field it's correct compared to the respective Business Rules involved. Do we need to use the .efx templates also in this case or they are needed only if we want to generate a UI for a caller client? If it is, how can we reach the goal?

Thank you for your help,
Matteo

**Comment by rouschr** [comment, marked as answer]
The central validation service (CVS) service is the validation tool for notice XML.
https://docs.ted.europa.eu/api/endpoints/cvs-ted-europa-eu.html
https://docs.ted.europa.eu/api/index.html
You need to setup a TED API key.
For your questions it is difficult to answer because I do not know what your product does.
I would say the CVS API reponse as SVRL is what you will have to focus on if you work directly with pre-filled XML notice files.
https://docs.ted.europa.eu/eforms/latest/guide/validation.html#_using_the_central_validation_service
